### DISCUSSION TITLE
setStore force replace

### DISCUSSION THREAD

**Discussion by odama626** [solidjs/solid#2298] [Ideas]
I ran into a use case where I want to be able to replace the value of a store and realized it isn't possible with the current api since using objects always merge, is there a recommended way to replace that?

https://playground.solidjs.com/anonymous/e4d159cc-294c-45f8-b1da-01a5ef93f78f

**Comment by atk** [comment]
```js
batch(() => {
  setStore('toBeOverwritten', undefined);
  setStore('toBeOverwritten', { newStuff: true });
});
```

**Comment by odama626** [comment]
ah. yep that'll do it, so simple I didn't think of it. thanks

**Comment by ryansolid** [comment]
Just go up a level. It only merges shallow.

```js
// instead of
setStore("key", newObject)

// do this
setStore({
  key: newObject
})
```

That being said we can't replace top level as we have an object proxy which needs a stable reference. We'd need to diff all the properties. `reconcile` can do this.
